src/main/java/seedu/address/logic/commands/SearchPatientCommand.java

```diff
@@ -0,0 +1,197 @@
+
+    private final SearchPatientDescriptor searchPatientDescriptor;
+    private final int areaNotFound = 0;
+    private final int areaIsName = 1;
+    private final int areaIsTemperature = 2;
+    private final int tooManyArea = 3;
+
+    /**
+     * Constructs an SearchPatientCommand to edit the patient with the name {@code String}.
+     *
+     * @param searchPatientDescriptor Description of the searchPerson command.
+     */
+    public SearchPatientCommand(SearchPatientDescriptor searchPatientDescriptor) {
+        requireNonNull(searchPatientDescriptor);
+        this.searchPatientDescriptor = searchPatientDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        final int areaToSearch = this.confirmArea(searchPatientDescriptor);
+        ArrayList<Person> personWithinTemperatureRange = new ArrayList<>();
+        List<Person> personList = model.getFilteredPersonList();
+
+        if (areaToSearch == areaNotFound) {
+            throw new CommandException(MESSAGE_NOT_FOUND);
+        }
+
+        else if (areaToSearch == areaIsName) {
+            String nameToSearch = searchPatientDescriptor.getName().toString().trim().toLowerCase();
+            Person personFound = personList.get(0);
+            boolean isPersonFound = false;
+
+            for (Person person : personList) {
+                String personName = person.getName().toString().trim().toLowerCase();
+                if (personName.equals(nameToSearch)) {
+                    personFound = person;
+                    isPersonFound = true;
+
+                }
+            }
+            if (!isPersonFound) {
+                throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+            } else {
+                return new CommandResult(String.format(MESSAGE_SEARCH_PERSON_SUCCESS, personFound));
+            }
+        }
+
+        else if (areaToSearch == areaIsTemperature) {
+            double startingTemperature = searchPatientDescriptor.getTemperatureRange().getStartingTemperature();
+            double endingTemperature = searchPatientDescriptor.getTemperatureRange().getEndingTemperature();
+            for (Person person : personList) {
+                if (person.getTemperature().getValue() >= startingTemperature
+                        && person.getTemperature().getValue() <= endingTemperature) {
+                    personWithinTemperatureRange.add(person);
+                }
+            }
+
+            if (personWithinTemperatureRange.isEmpty()) {
+                throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+            } else {
+                return new CommandResult(MESSAGE_SEARCH_PERSON_LIST_SUCCESS
+                        + getListOutput(personWithinTemperatureRange));
+            }
+        } else if (areaToSearch == tooManyArea) {
+            return new CommandResult(Messages.MESSAGE_TOO_MANY_COMMANDS);
+        }
+        return null;
```

Comment:
I'm not too sure but I think the execution will not reach this return statement? If that is the case, I think we can throw a `CommandException` instead of returning `null`? Returning `null` here may cause a `NullPointerException` somewhere else.

Reply:
Definitely true